modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ANode.java

```diff
@@ -44,6 +44,24 @@
  */
 public abstract class ANode {
 
+    protected ANode parent;
+
+    public ANode getParent() {
+        return parent;
+    }
+
+    public List<ANode> getChildren() {
```

Comment:
I thought we had moved away from trying to have all nodes have some generic iteration/manipulation of children?